doc/build-helpers/build-helpers.chapter.md

```diff
@@ -0,0 +1,106 @@
+This is not to be confused with the [`builder` argument of the Nix `derivation` primitive](https://nixos.org/manual/nix/unstable/language/derivations.html), which refers to the executable that produces the build result, or [remote builder](https://nixos.org/manual/nix/stable/advanced-topics/distributed-builds.html), which refers to a remote  machine that could run such an executable.
+:::
+
+Such a function is usually designed to abstract over a typical workflow for a given programming language or framework.
+This allows declaring a build recipe by setting a limited number of options relevant to the particular use case instead of using the `derivation` function directly.
+
+[`stdenv.mkDerivation`](#part-stdenv) is the most widely used build helper, and serves as a basis for many others.
+In addition, it offers various options to customize parts of the builds.
+
+There is no uniform interface for build helpers.
+[Trivial build helpers](#chap-trivial-builders) and [fetchers](#chap-pkgs-fetchers) have various input types for convenience.
+[Language- or framework-specific build helpers](#chap-language-support) usually follow the style of `stdenv.mkDerivation`, which accepts an attribute set or a fixed-point function taking an attribute set.
+
+## Fixed-point arguments of build helpers {#sec-build-helpers-finalAttrs}
+
+As mentioned above, `stdenv.mkDerivation` could alternatively accept a fixed-point function. The input of such function, typically named `finalAttrs`, is expected to be the final state of the attribute set.
+A build helper like this is said to accept **fixed-point arguments**.
+
+As [`stdenv.mkDerivation`'s support of fixed-point arguments](#mkderivation-recursive-attributes) was first included in the Nixpkgs release 22.05, `mkDerivation`-style build helpers don't always come with such support yet.doc/nixos
+
+### Defining a build helper with `lib.extendMkDerivation` {#ssec-build-helper-extendMkDerivation}
+
+The Nixpkgs library function `lib.customisation.extendMkDerivation` can be used to define a build helper with fixed-point arguments support from an existing one. It can also bring fixed-point argument support to existing builders as a slight wrapper around the original definition.
+
+The following is the conventional definition of build helper `mkLocalDerivation` that doesn't support fixed-point arguments:
+
+:::{.example #ex-build-helpers-mkLocalDerivation-plain}
+
+# Example definition of `mkLocalDerivation` without fixed-point arguments support
+
+Write the followings in `pkgs/path/to/local-derivation/default.nix`:
+
+```nix
+{ lib
+, stdenv
+}:
+
+{ preferLocalBuild ? true
+, allowSubstitute ? false
+, impassablePredicate ? (_: false)
+}@args:
+
+stdenv.mkDerivation (removeAttrs args [
+  "impassablePredicate"
+] // {
+  inherit
+    preferLocalBuild
+    allowSubstitute
+    ;
+})
+```
+
+and call the definition in [`all-packages.nix`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix):
```

Comment:
Since this is part of the rendered user docs, this should not reference Nixpkgs-internal files only relevant for contributions. Instead this should just avoid the reference to `all-packages.nix`.

Reply:
Addressed.